### ISSUE TITLE
When iVim crashes on startup: need for `vim -u NONE`

### ISSUE THREAD

**Issue by bwhelm** [terrychou/iVim#9]
I mentioned in #8 that long global variable names will crash iVim immediately. When this happens on startup because the variable name is in `.vimrc`, iVim becomes unusable and must be re-installed, and any customizations in `.vim` must be recreated.

It would be nice, therefore, to have the possibility of something equivalent to `vim -u NONE` so that the user can recover without starting over. One possibility here would be to have an option in iVim's entry in the Settings.app such that when toggled iVim will launch without .vimrc or plugins on the next launch: the equivalent of `vim -u NONE`.

**Comment by terrychou** [comment]
This is a good idea. Actually, I have already added options in the Settings.app for users to offer launch arguments for iVim, in the uncommitted code. I don't know how you are using iVim: install it from the App Store or compile it yourself? If the latter, I can push the code to github so that you can test before I put it on the App Store?

**Comment by bwhelm** [comment]
I don't have a paid Apple account, so I've installed iVim from the App
Store. I can compile it to test (albeit without sharing capabilities). It's
also possible to use TestFlight.app for testing. Whatever's easiest for you.

**Comment by terrychou** [comment]
Let's try the compiling way this time, because the compiling way itself needs testing too :).

**Comment by bwhelm** [comment]
Well, I can sort of test it. I have iOS 11 public beta 5 installed on my iPad Air 2. When I try compiling from Xcode 8.3.3, Xcode tells me it "may not be compatible" with iOS 11, and so it refuses to compile. So I can't test it live.

However, I did use the simulator, testing on an iPad Air 2. Here's what I did:

1. Launch iVim, create `~/.vim/vimrc` with the following:

    > let g:aVeryLongGlobalVariableNameToMakeVimCrashOnLaunch = 1
    > echom 'Testing'

As reported in #8, this currently crashes iVim when sourced. Quitting iVim and relaunching results in the expected crash.

2. I then go into Preferences.app, enter `-u NONE` into the iVim => Launch Options => Arguments field, relaunch iVim, and it does not crash; entering `:messages` results in nothing being displayed. All of this is as expected.

3. I then comment out the first line, save, quit, and relaunch iVim. Now it does not crash, and `:messages` results in `Testing` being displayed, again as expected.

In short, using the simulator, everything seems to work well. The UI in Preferences.app seems adequate, but I wonder if it would be better to have an additional note saying something like "To launch iVim without sourcing $VIMRC or plugins, use `-u NONE`." After all, if users cannot launch iVim, they won't have access to vim's help files to know that `-u NONE` is the appropriate argument to use in this case.

**Comment by terrychou** [comment]
Thank you! It seems that the launch arguments part is working, though not all of the arguments are supported here.

As to the additional note, although it may help in some cases, it looks a little bit redundant in most other ones. My opinion is since the user has already come to this point turning to the command line arguments, they must have known which ones they want.

However, it may be better to offer some simpler options, such as "launch with factory default":D, for beginners to use. what do you say?

**Comment by bwhelm** [comment]
"Launch with factory defaults" sounds good to me. It's not only beginners who will need a reminder of what the command line arguments are: I had to look up `-u NONE`, and in 6 months when I need it again I'll probably have forgotten.

**Comment by sedm0784** [comment]
I'd prefer "Launch with Vim defaults" and for the arguments used to be `-Nu NONE`. I occasionally like to ignore my `vimrc`. I never want to use `vi` defaults.
